modules/gdscript/gdscript_parser.cpp

```diff
@@ -4092,34 +4141,28 @@ bool GDScriptParser::export_annotations(const AnnotationNode *p_annotation, Node
 	DataType export_type = variable->get_datatype();
 	bool use_default_variable_type_check = true;
 
+	// Process array and packed array annotations on the element type.
+	bool is_array = false;
+	if (export_type.builtin_type == Variant::ARRAY && export_type.has_container_element_type(0)) {
+		is_array = true;
+		export_type = export_type.get_container_element_type(0);
+	} else if (export_type.is_typed_container_type()) {
+		is_array = true;
+		export_type = export_type.get_typed_container_type();
+	}
```

Comment:
```suggestion
	bool is_array = false;
	if (p_annotation->name != SNAME("@export")) {
		if (export_type.builtin_type == Variant::ARRAY && export_type.has_container_element_type(0)) {
			is_array = true;
			export_type = export_type.get_container_element_type(0);
		} else if (export_type.is_typed_container_type()) {
			is_array = true;
			export_type = export_type.get_typed_container_type();
		}
	}
```

Reply:
@dalexeev Actually, this solution breaks multiple tests:

- export_variable: `@export var test_node_array: Array[Node]`
- export_node_in_non_node_derived_class_3

I think I may have found a different solution for that.

Reply:
See a suggestion below.

Reply:
Ignoring this (since otherwise @export with arrays breaks) but applying the type_source suggestion fixed all issues.

Reply:
@dalexeev 
Gah, GitHub did show me some added suggestions but not your later comments. I replied to this without seeing your latest comment. Sorry about that lol